---
title: Effect size of a tag variant (binary outcome)
author: ''
date: '2020-11-01'
draft: true
slug: effect-size-of-a-tag-variant-binary-outcome
categories: [derivation]
tags:
  - effect size
  - LD
  - tag variant
subtitle: ''
summary: ''
authors: [admin]
lastmod: '2020-11-01T12:54:03-05:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---

#### Motivation

Recently, I've been studying the effect of population stratification on GWAS effect size estimates. For this purpose, I needed to work out the effect size of a tag SNP as a function of its LD with the causal variant. I've always taken for granted that this is equal to $r\beta_{causal}$, where $r$ is the LD between the two variants and $\beta_{causal}$ is the causal effect size. This makes intuitive sense qualitatively because we expect tag SNPs that are further away from causal variants to have smaller effect sizes (and vice versa). But I couldn't intuit why this exact relationship. There are some derivations for this in the early GWAS literature (e.g. see [^1]) but none that were very explicit so I decided to work it out and post in case others find it helpful. 

Side warning: I would not recommend reading this on a smaller screen (e.g. phone) as the math might not render fully. 

#### Modeling the effect of causal variant on disease risk

Suppose there are two biallelic loci, A and B, where A is the causal locus (alleles at this locus have a causal effect on the phenotype) and B is a biallelic SNP in LD with A. 

Let:

* $Y \in \{0,1\}$ indicate disease status, where 0 indicates being healthy and 1 indicates having the disease.

* $A \in \{0,1\}$ indicate whether or not a haplotype carries the non-reference (risk) allele at the causal variant A. Note that the terms 'reference' and 'non-reference' are arbitrary and are introduced only to clarify which alleles I am talking about.

We can model the effect of the causal variant on disease risk using a logistic regression model:

$$log \left( \frac{\pi}{1-\pi} \right) =\mu + \beta_A G + e$$

where the left hand side is the log odds of developing the disease, $G$ represents the number of risk alleles, $\beta_A$ represents the effect size, and $e$ is some random error.

We can measure the effect of the risk allele in terms of its penetrances, i.e., the probability that an individual develops the disease given that they carry the allele. 

Let the penetrance of the alleles at the causal locus be:

$$\begin{aligned}
P(Y = 1|A = 0) = a_0,~~ and ~~P(Y = 1|A = 1) = a_1
\end{aligned}$$ 

Note, that $a_0 = \mu + e$ and $a_1 = \mu + \beta_A + e$. Then, assuming an additive model (i.e. no dominance; additional risk alleles increase the risk by the same amount), we can get the effect size $\beta_A$ by subtraction:

$$\begin{aligned}
& P(Y = 1|A = 1) - P(Y = 1|A = 0) \\
= & a_1 - a_0 \\
= &  \mu + \beta_A + e - \mu -e \\
= & \beta_A
\end{aligned}$$

Similarly, we can write down the effect size of the marker locus. Let the penetrances be $P(Y = 1|B = 0) = b_0$ and $P(Y = 1|B = 1) = b_1$. Then the effect size is:

$$\beta_B = b_0 - b_1$$

Our goal is to answer the question: what is the relationship between the effect of a tag SNP and the effect of a causal variant given the LD between them? To do so, let's first write down the LD between the two variants.

#### Modeling the LD between locus A and B

We can represent the LD as the correlation between the two loci:

$$r = \frac{ f_{AB} - f_A f_B }{\sqrt{f_A (1-f_A) f_B (1-f_B)}}$$

where $f_A = P(A = 1)$ and $f_B = P(B = 1)$ are the allele frequencies at locus A and B, respectively, and $f_{AB}$ is the frequency of the haplotype carrying non-reference alleles at both loci (i.e. $f_{AB} = P(A = 1,B = 1)$). We need to further simplify $f_{AB}$ in terms of the conditional probability of observing the risk allele at locus A given that we know the allele at B.

Let $P(A = 1|B = 0) = q_0$ and $P(A = 1|B = 1) = q_1$

Then, the frequency of the $f_{AB}$ haplotype is: 

$$\begin{aligned}
f_{AB} = & P(A = 1,B = 1) \\
= & P(A = 1|B = 1)P(B = 1) \\
= & q_1 f_B
\end{aligned}$$

We can similarly express the frequency of other haplotypes. For example: 

$$\begin{aligned}
f_{Ab} = & P(A = 1,B = 0) \\
= & P(A = 1|B = 0)P(B = 0) \\
= & q_0 (1 - f_B)
\end{aligned}$$

Substituting $f_{AB}$ in the numerator of $r$, we get:

$$f_{AB} - f_A f_B = q_1 f_B - f_A f_B$$

We can further simplify this by expressing $f_A$ in terms of $f_B$ by marginalization.

$$\begin{aligned}
& f_A = P(A = 1) \\
& = P(A = 1,B = 1) + P(A = 1,B = 0) \\
& = q_1f_B + q_0(1 - f_B)
\end{aligned}$$

Then, 

$$\begin{aligned}
f_{AB} - f_A f_B & = f_Bq_1 - f_B \{q_1f_B + q_0(1 - f_B)\} \\
& = f_B(1 - f_B)(q_1 - q_0)
\end{aligned}$$

and $r$ becomes:

$$\begin{aligned}
r & = \frac{f_B(1 - f_B)(q_1 - q_0)}{\sqrt{f_B(1-f_B)}\sqrt{f_A(1-f_A)}} \\
& = (q_1 - q_0)\frac{\sqrt{f_B(1-f_B)}}{\sqrt{f_A(1-f_A)}}
\end{aligned}$$


#### Modeling the effect of the tag variant (B) on disease risk

In our model, the tag variant (B) itself has no inrinsic effect on disease risk. Its effect size is solely due to its correlation with the causal variant (A). Thus, the risk for any haplotype carrying the reference allele at locus B depends on which allele is present at locus A. So, to derive the expected effect size at the tag variant, we need to marginalize over the allele at the causal variant. 

The penetrance of the reference allele at the tag variant can be written as: 

$$\begin{aligned}
& P(Y = 1|B = 0) = \frac{P(Y = 1,B = 0)}{P(B = 0)}
\end{aligned}$$

To calculate this, marginalize over the allele at the causal variant.

$$\begin{aligned}
& P(Y = 1, B = 0) \\
= & P(Y = 1, B=0|A = 0)P(A = 0) +\\
= & P(Y = 1, B=0|A = 1)P(A = 1)
\end{aligned}$$

The effect of B on $Y$ are independent (conditional on A). Thus,

$$\begin{aligned}
& P(Y = 1,B = 0|A = 0)P(A = 0) \\
= & P(Y = 1|A = 0)P(B = 0|A = 0)P(A = 0) \\ 
= & a_0 \frac{P(B = 0, A = 0)}{P(A = 0)}P(A = 0) \\
= & a_0(1-q_0)(1-f_B)
\end{aligned}$$

Similarly,

$$\begin{aligned}
& P(Y = 1,B = 0|A = 1)P(A = 1) \\
= & P(Y = 1|A = 1)P(B = 0|A = 1)P(A = 1) \\
= & a_1 \frac{P(B = 0, A = 1)}{P(A = 1)}P(A = 1) \\
= & a_1 q_0 (1-f_B)
\end{aligned}$$

Therefore, $P(Y = 1, B = 0)$ becomes:

$$\begin{aligned}
P(Y = 1,B = 0) & = a_0(1-q_0)(1-f_B) + a_1 q_0 (1-f_B)\\
& = (1 - f_B) \{a_0(1-q_0) + a_1q_0 \}
\end{aligned}$$

And $P(Y = 1|B = 0)$ is: 

$$\begin{aligned}
P(Y = 1|B = 0) & = (1 - f_B) \frac{a_0(1-q_0) + a_1q_0}{1-f_B} \\
= & a_0(1-q_0) + a_1q_0
\end{aligned}$$

We can do the same thing for $b_1 = P(Y = 1|B = 1)$:

$$\begin{aligned}
& P(Y = 1|B = 1) \\
= & \frac{P(Y = 1, B = 1)}{P(B = 1)} \\
= & P(Y = 1, B = 1|A = 0)P(A = 0) + \\
& \frac{P(Y = 1, B = 1|A = 1)P(A = 1)}{P(B = 1)} \\
= & P(Y = 1| A = 0)P(B = 1|A = 0)P(A = 0) + \\
 & \frac{P(Y = 1|A = 1)P(B = 1|A = 1)P(A = 1)}{P(B = 1)} \\
= & \frac{a_0(1 - q_1)f_B + a_1q_1f_B}{f_B} \\
= &a_0(1 - q_1) +a_1q_1
\end{aligned}$$

The effect size of B ($\beta_B$) is (similar to how we calculated the effect size of A):

$$\begin{aligned}
& P(Y = 1|B = 1) - P(Y = 1|B = 1) \\
= & b_1 - b_0 \\
= & a_0(1 - q_0) + a_1(1 - q_0) - \{a_0(1 - q_1) + a_1q_1\} \\
= & a_0 - a_0q_0 + a_1q_0 - a_1q_1 - a_0 + a_0q_1 \\
= & q_0(a_1 - a_0) + q_1(a_1 - a_0) \\
= & (q_1 - q_0)(a_1 - a_0) \\
= & (q_1 - q_0)\beta_A
\end{aligned}$$

We know from earlier that:

$$r = (q_1 - q_0)\frac{\sqrt{f_B(1-f_B)}}{\sqrt{f_A(1-f_A)}}$$

Then, 

$$q_1 - q_0 = r \frac{\sqrt{f_A(1-f_A)}}{\sqrt{f_B(1-f_B)}}$$

And:

$$\beta_B = \beta_A r\frac{\sqrt{f_A(1-f_A)}}{\sqrt{f_B(1-f_B)}}$$

In practice, the genotypes are normalized to unit variance so $f_B(1-f_B) = f_A(1-f_A)$. If that's the case, $\beta_B$ further simplifies to:

$$\beta_B = r\beta_A$$

Done! Although, note that this simplified relationship assumes that genotypes for all SNPs are normalized to the same variance.

I was going to add further derivations here for the standard error  of the effect size and power of the test as a function of $r$ but realize it's already a bit long and leave these for a future post. 

Cheers! :)

[^1]: adapted from [Vukcevic et al. 2012](https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.20576)

